include/fmt/format-inl.h

```diff
@@ -968,16 +969,27 @@ FMT_FUNC void report_windows_error(int error_code,
 #endif
 
 FMT_FUNC void vprint(std::FILE* f, string_view format_str, format_args args) {
+  if (std::fwide(f, -1) >= 0) {
+    FMT_THROW(system_error(EILSEQ,
+              "cannot write byte data into wide-oriented stream"));
+  }
   memory_buffer buffer;
   internal::vformat_to(buffer, format_str,
                        basic_format_args<buffer_context<char>>(args));
   internal::fwrite_fully(buffer.data(), 1, buffer.size(), f);
 }
 
 FMT_FUNC void vprint(std::FILE* f, wstring_view format_str, wformat_args args) {
+  if (std::fwide(f, 1) <= 0) {
```

Comment:
Doesn't `fputws` have this check already?

Reply:
Yes, similar to `fwrite`, `fputws` will return a failure condition if the orientation is incorrect. I figured an early bail-out with a more friendly exception message would be worth adding.

Reply:
and this